README.rst

```diff
@@ -114,7 +114,7 @@ Some `purl` examples
     pkg:deb/debian/curl@7.50.3-1?arch=i386&distro=jessie
 
     pkg:docker/cassandra@sha256:244fd47e07d1004f0aed9c
-    pkg:docker/gcr.io/customer/dockerimage@sha256:244fd47e07d1004f0aed9c
+    pkg:docker/customer/dockerimage@sha256:244fd47e07d1004f0aed9c?repository_url=gcr.io
```

Comment:
As I read the spec it should be perfectly fine to let the 'namespace' part of the purl be 'gcr.io/customer' as in the example already given. Do you read it differently? https://github.com/package-url/purl-spec#rules-for-each-purl-component

Reply:
According to the '[Docker Registry HTTP API V2](https://docs.docker.com/registry/spec/api/#overview)' definition of a name can be composed from one or more component paths.
When parsing a docker package definition it's not clear if there is an repository present or not.

Does the first part after `pkg:docker/<optional_repository>` overwrite the default repository name? If so, what is the rule? If 3 or more path component parts are present, then consider the first element as repository. This only works if every implementation of the 'Docker Registry HTTP API V2' limits the name to two path components.

E.g:
This is a possible namespace without repository declaration
pkg:docker/firstpart/secondpart/thirdpart/dockerimage@sha256:244fd47e07d1004f0aed9c

I hope that clears up the pull request.